File: Data Structures & Algorithms/products-of-array-discluding-self/submission-5.cs
public class Solution
{
    public int[] ProductExceptSelf(int[] nums)
    {
        var newArr = new int[nums.Length];
        newArr[0]= 1;

        for (int i = 1; i < nums.Length; i++)
        {
            newArr[i] = nums[i - 1] * newArr[i - 1];
        }

        var suff = 1;
        for (int i = nums.Length - 2; i >= 0; i--)
        {
            suff *= nums[i + 1];
            newArr[i] *= suff;
        }
        
        return newArr;
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-5.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_set<int> s;
        for (int i = 0; i < nums.size(); i++) {
            // 1. Kiểm tra xem số hiện tại đã xuất hiện trong khoảng k chưa
            if (s.find(nums[i]) != s.end()) return true;
            
            // 2. Thêm số hiện tại vào set
            s.insert(nums[i]);
            
            // 3. Nếu kích thước set vượt quá k, xóa phần tử xa nhất (bên trái)
            if (s.size() > k) {
                s.erase(nums[i - k]);
            }
        }
        return false;
    }
};